nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -296,6 +296,15 @@ def fwd_bwd_step(self, dataloader_iter, batch_idx, forward_only):
         tensor_shape = [seq_length, get_micro_batch_size(), self.cfg.hidden_size]
         data_iter = get_iterator_k_split(batch, get_num_microbatches())
 
+        # handle asynchronous grad reduction
+        no_sync_func = None
+        grad_sync_func = None
+        param_sync_func = None
+        if not forward_only and self.with_distributed_adam:
+            no_sync_func = partial(self._optimizer.no_sync, greedy_grad_copy=self.megatron_amp_o2,)
```

Comment:
@MaximumEntropy Looks like we forgot `from functools import partial`

Reply:
Ooops can you approve this? https://github.com/NVIDIA/NeMo/pull/7026

Reply:
nice catch! approved the import PR